### PR TITLE
Tb user stories 34 and 35

### PR DESCRIPTION
**Pull request by TravisBorgsmiller**
- Has alexis code from his PR user story 29 
- Added test for merchant dashboard to see merchant info and pending order information
- Added view information and controller logic for merchant dashboard
- changed route for admin and merchant dashboard to equal what matches in the user stories 
- changed expected routes in some tests to account for the new merchant and admin dashboard links
- changed routes file to match

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@ class Merchant <ApplicationRecord
   has_many :items, dependent: :destroy
   has_many :item_orders, through: :items
   has_many :users
-
+  has_many :orders, through: :item_orders
```
Comment (adumortier) [inline app/models/merchant.rb:5]: I don't see a `belongs_to` relationship in the `order` model. did I miss something?

File: app/models/order.rb [hunk 2]
```diff
@@ -15,4 +15,12 @@ def grandtotal
   def total_quantity
     items.sum(:quantity)
   end
+
+  def item_quantity_merchant(merchant)
+    items.where(merchant: merchant).sum("item_orders.quantity")
```
Comment (adumortier) [inline app/models/order.rb:20]: AR rocks ! :)

**Review by adumortier — APPROVED** [review]

**Comment by adumortier** [comment]
Haven't tested in `rails s` yet but all tests passed.

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -31,9 +31,9 @@ def login(user)
       if user.default?
         redirect_to '/profile'
       elsif user.merchant_employee?
-        redirect_to '/merchant/dashboard'
+        redirect_to '/merchant'
       elsif user.admin?
-        redirect_to '/admin/dashboard'
+        redirect_to '/admin'
       end
```
Comment (cgaddis36) [inline app/controllers/sessions_controller.rb:37]: I like the new routes

File: spec/features/orders/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      expect(page).to have_content(order1.city)
+      expect(page).to have_content(order1.state)
+      expect(page).to have_content(order1.zip)
+
+      within("section#item-#{@item1.id}") do 
+        expect(page).to have_content(@item1.name)
+        expect(page).to have_content(@item1.description)
+        expect(page).to have_content(2)
+        expect(page).to have_content("#{ActiveSupport::NumberHelper.number_to_currency(@item1.price)}")
+        expect(page).to have_content("#{ActiveSupport::NumberHelper.number_to_currency(@item1.price * 2)}")
```
Comment (cgaddis36) [inline spec/features/orders/show_spec.rb]: I believe these are Alexs tests?

File: spec/features/users/visitor_authorization_spec.rb [hunk 5]
```diff
@@ -2,10 +2,10 @@
 
 RSpec.describe "As a visitor", type: :feature do
   it "Can't go to unauthorized pages" do
-    visit '/admin/dashboard'
+    visit '/admin'
```
Comment (cgaddis36) [inline spec/features/users/visitor_authorization_spec.rb:5]: Nice edge cases here!

File: app/views/merchant/dashboard/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+<%= current_user.merchant.state %>
+<%= current_user.merchant.zip %></p>
+
+<section class="grid-container">
+
+<% @orders.each do |order| %>
+<% if order.status == "pending" %>
+<section class="grid-item" id = "pending-<%= order.id %>">
+  <p>ID: <%= link_to "#{order.id}", "/merchant/orders/#{order.id}" %></p>
+  <p>Order Created: <%= order.created_at %></p>
```
Comment (cgaddis36) [inline app/views/merchant/dashboard/index.html.erb:15]: Nice job looks good here

**Review by KellyIB — APPROVED** [review]

**Pull request merged**